packages/@lwc/ssr-compiler/src/compile-js/decorators/wire.ts

```diff
@@ -190,7 +190,10 @@ const bCallWiredMethod = esTemplate`
 `<ExpressionStatement>;
 
 const bWireAdapterPlumbing = esTemplate`{
-    const wireInstance = new ${/*wire adapter constructor*/ is.expression}((newValue) => {
+    // Callable adapters are expressed as a function having an 'adapter' property, which
+    // is the actual wire constructor.
+    const AdapterCtor = ${/*wire adapter constructor*/ is.expression}.adapter ?? ${/*wire adapter constructor*/ 0};
```

Comment:
What happens if someone does `@wire(somethingNullish)`? It'd break, obviously, but how catastrophically?

Reply:
Good thought, I'll use optional chaining.